go/ros/bag2mcap.go

```diff
@@ -202,6 +203,12 @@ func processBag(
 				} else {
 					chunkReader.Reset(r)
 				}
+			case "bz2":
+				if chunkReader == nil {
+					chunkReader = bzip2.NewReader(r)
```

Comment:
I think you will need to use,

```
&resettableByteReader{bzip2.NewReader(r)}
```

here to make it conform to the expected interface. The bzip2 reader in the compress package doesn't actually support Resetting - this will simply fake it.